Main/ClientApp/src/Components/CollectionView.tsx

```diff
@@ -0,0 +1,57 @@
+import { RouteComponentProps } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import CollectionsStore from "../Stores/CollectionsStore";
+import { Collection } from "../Model/Collection";
+import ElementView from "./ElementView";
+import { GridList, GridListTile } from "@material-ui/core";
+import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    list: {
+      padding: "25px 30px 25px 30px"
+    }
+  })
+);
+
+type CollectionViewProps = RouteComponentProps & {
+  match: { params: { collectionId: number } };
+};
+
+const canRenderExtra = (url: string) => {
+  // let regexp : RegExp = /\.(jpeg|jpg|gif|png)$/;
+  return url.match(/\.(jpeg|jpg|gif|png)$/) != null;
```

Comment:
For now I guess "extra" only refers to image links, but that will not be the case in the future. This way of implementing is not future proof and will need to be rewritten when we add new functionality. 
Consider adding some class that will get the type of it and return proper component (ImageElement, VideoElement or something like that)

Reply:
Good point, also this whole gridList needs to be rewritten because it does not look decently.